class.jetpack-autoupdate.php

```diff
@@ -48,5 +59,68 @@ function autoupdate_core( $update, $item ) {
 		}
 		return $update;
 	}
+
+	function automatic_updates_complete( $results ) {
+		$this->autoupdate_results = $results;
+	}
+
+	private function get_successful_updates( $results = 'plugin' ) {
+		$successful_updates = array();
+
+		if ( ! isset( $this->autoupdate_results[ $results ] ) ) {
+			return $successful_updates;
+		}
+
+		foreach( $this->autoupdate_results[ $results ] as $result ) {
+			if ( $result->result ) {
+				switch( $result ) {
+					case 'theme':
+						$successful_updates[] = $result->item->theme;
+						break;
+					default:
+						$successful_updates[] = $result->item->slug;
+				}
+			}
+		}
+
+		return $successful_updates;
+	}
+
+	function log_results() {
+
+		// if we've run an update, lets compare the expected results to the actual results, and log our findings
+		if( $this->is_updating === true  ) {
+
+			$plugins_updated = 0;
+			$plugins_failed  = 0;
+			$plugin_results  = $this->get_successful_updates();
+
+			foreach( $this->autoupdate_expected['plugin'] as $plugin ) {
+				if ( in_array( $plugin->slug, $plugin_results ) ) {
+					$plugins_updated++;
+				} else {
+					$plugins_failed++;
+				}
+			}
+
+			$stats = array();
+
+			if ( $plugins_updated ) {
+				$stats['x_jetpack_autoupdates/plugin-success'] = $plugins_updated;
```

Comment:
Please don't manually create it.  This line should be `$jetpack->stat( 'autoupdates/plugin-success', implode( ',', $plugins_updated ) );` with some way to access $jetpack -- perhaps a `$this->jetpack = Jetpack::init()` in the constructor or higher up in the function.